workflow/initiator_group_test.go

```diff
@@ -0,0 +1,144 @@
+	dupIG := &nimbleos.InitiatorGroup{
+		Name:           param.NewString(initiatorGroupName),
+		Description:    param.NewString("Workflow initiator group"),
+		AccessProtocol: nimbleos.NsAccessProtocolIscsi,
+	}
+	_, err := suite.initiatorGrpService.CreateInitiatorGroup(dupIG)
+	assert.NotNil(suite.T(), err, "Initiator group creation should have failed")
+}
+
+func (suite *IGWorkflowSuite) TestCreateIGIscsiInitiators() {
+	iscsiInitiator := &nimbleos.NsISCSIInitiator{
+		Label:     param.NewString("initiator1"),
+		Iqn:       param.NewString("iqn.1994-05.com.redhat:48e040d53bf6"),
+		IpAddress: param.NewString("10.0.0.0"),
+	}
+	var initiatorList []*nimbleos.NsISCSIInitiator
+	initiatorList = append(initiatorList, iscsiInitiator)
+	newIG := &nimbleos.InitiatorGroup{
+		Name:            param.NewString("TestIGIscsi"),
+		Description:     param.NewString("Workflow initiator group"),
+		AccessProtocol:  nimbleos.NsAccessProtocolIscsi,
+		IscsiInitiators: initiatorList,
+	}
+	_, err := suite.initiatorGrpService.CreateInitiatorGroup(newIG)
+	assert.Nilf(suite.T(), err, "Unable to create initiator group, err: %v", err)
+}
+
+func (suite *IGWorkflowSuite) TestCreateIGInvalidIscsi() {
+	iscsiInitiator := &nimbleos.NsISCSIInitiator{
+		Label:     param.NewString("initiator1"),
+		Iqn:       param.NewString("a"),
+		IpAddress: param.NewString("10.0.0.0"),
+	}
+	var initiatorList []*nimbleos.NsISCSIInitiator
+	initiatorList = append(initiatorList, iscsiInitiator)
+	newIG := &nimbleos.InitiatorGroup{
+		Name:            param.NewString("TestIGIscsiInvalid"),
+		Description:     param.NewString("Workflow initiator group"),
+		AccessProtocol:  nimbleos.NsAccessProtocolIscsi,
+		IscsiInitiators: initiatorList,
+	}
+	_, err := suite.initiatorGrpService.CreateInitiatorGroup(newIG)
+	assert.NotNil(suite.T(), err, "Initiator group cretaion with invalid iscsi initiators should have failed")
+}
+
+func (suite *IGWorkflowSuite) TestModifyIGDescription() {
+	var desc *string = param.NewString("Workflow tests for initiator group")
+	updateIG := &nimbleos.InitiatorGroup{
+		Description: desc,
+	}
+	ig, _ := suite.initiatorGrpService.GetInitiatorGroupByName(initiatorGroupName)
+	if ig != nil {
+		suite.initiatorGrpService.UpdateInitiatorGroup(*ig.ID, updateIG)
+	}
+	ig, _ = suite.initiatorGrpService.GetInitiatorGroupByName(initiatorGroupName)
+	assert.Equalf(suite.T(), *desc, *ig.Description, "Unable to update Initiatot Group, err: %v", ig.Description)
+}
+
+func (suite *IGWorkflowSuite) TestModifyIGInitiators() {
+	var ipAdd *string = param.NewString("10.1.0.0")
+	iscsiInitiator := &nimbleos.NsISCSIInitiator{
+		IpAddress: ipAdd,
+		Label:     param.NewString("NewLabel"),
+	}
+	var initiatorList []*nimbleos.NsISCSIInitiator
+	initiatorList = append(initiatorList, iscsiInitiator)
+	updateIG := &nimbleos.InitiatorGroup{
+		IscsiInitiators: initiatorList,
+	}
+	currentIG, _ := suite.initiatorGrpService.GetInitiatorGroupByName("TestIGIscsi")
```

Comment:
I wonder if each test case should be self-contained, rather than relying on the result from another test case.  For example, this test case could perform a Create, Modify, and Delete.

Reply:
Makes sense. Will take care of it in my next PR.